File: examples/spacetrack/generic_request.py
# /// script
# dependencies = ["brahe"]
# ///
"""
Demonstrates making generic SpaceTrack API requests.
"""

import os

import brahe as bh

# Get credentials from environment
# Note: For production use, most users should use the default constructor:
#   client = bh.SpaceTrackClient(user, password)
# These examples use the test server for automated testing.
user = os.environ.get("TEST_SPACETRACK_USER", "")
password = os.environ.get("TEST_SPACETRACK_PASS", "")

if not user or not password:
    print("Set TEST_SPACETRACK_USER and TEST_SPACETRACK_PASS environment variables")
else:
    client = bh.SpaceTrackClient(
        user, password, base_url="https://for-testing-only.space-track.org/"
    )

    # Make a generic request for custom queries
    # This returns raw response text
    response = client.generic_request(
        controller="basicspacedata",
        class_name="gp",
        predicates={"NORAD_CAT_ID": "25544", "limit": "1"},
    )

    print(f"Response length: {len(response)} characters")
    print(f"Response preview: {response[:100]}...")
    # Response length: 1234 characters
    # Response preview: [{"CCSDS_OMM_VERS":"2.0",...

print("Example completed successfully!")

File: examples/spacetrack/generic_request.rs
//! Demonstrates making a generic SpaceTrack API request.
//!
//! FLAGS = ["CI-ONLY"]

use brahe::spacetrack::{BlockingSpaceTrackClient, TEST_BASE_URL, equals};

fn main() {
    // Get credentials from environment
    // Note: For production use, most users should use the default constructor:
    //   BlockingSpaceTrackClient::new("user", "password")
    // These examples use the test server for automated testing.
    let user = std::env::var("TEST_SPACETRACK_USER").unwrap_or_default();
    let password = std::env::var("TEST_SPACETRACK_PASS").unwrap_or_default();

    if user.is_empty() || password.is_empty() {
        println!("Set TEST_SPACETRACK_USER and TEST_SPACETRACK_PASS environment variables");
    } else {
        let client = BlockingSpaceTrackClient::with_base_url(&user, &password, TEST_BASE_URL)
            .expect("Failed to create client");

        // Make a raw API request with custom predicates
        let predicates = [
            ("NORAD_CAT_ID", equals("25544")),
            ("limit", equals("1")),
        ];

        let response = client
            .generic_request("basicspacedata", "gp", &predicates, None)
            .expect("Request failed");

        // Response is raw JSON string
        println!("Response length: {} bytes", response.len());
        println!("Contains ISS: {}", response.contains("ISS"));
        // Response length: 1234 bytes
        // Contains ISS: true
    }

    println!("Example completed successfully!");
}
